WooCommerce/Classes/Tools/Currency/CurrencyFormatter.swift

```diff
@@ -110,7 +110,13 @@ public class CurrencyFormatter {
     ///     - currency: a 3-letter country code for currencies that are supported in the API. e.g. "USD"
     ///
     func formatAmount(_ stringAmount: String, with currency: String = CurrencySettings.shared.currencyCode.rawValue) -> String? {
-        guard let decimalAmount = convertToDecimal(from: stringAmount) else {
+
+        // NSDecimalNumber use by default the local decimal separator to evaluate a decimal amount.
+        // We substitute the current decimal separator with the locale decimal separator.
+        let localeDecimalSeparator = Locale.current.decimalSeparator ?? CurrencySettings.shared.decimalSeparator
+        var newStringAmount = stringAmount.replacingOccurrences(of: ",", with: localeDecimalSeparator)
+        newStringAmount = newStringAmount.replacingOccurrences(of: ".", with: localeDecimalSeparator)
```

Comment:
Can these new changes be moved to the `convertToDecimal` method? It would be good to assume all strings that need converted to NSDecimalNumber will have this problem.

Reply:
You are totally right, thanks for noticing it! 🙇 
I updated the PR.